scripts/pylib/twister/twisterlib.py

```diff
@@ -749,10 +749,19 @@ def handle(self):
         ser_pty_process = None
         if serial_pty:
             master, slave = pty.openpty()
+            serial_cmd = re.split(',| ', serial_pty)
+
+            # We also pass the build directory to the --device-serial-pty. In some boards, for example,
+            # intel_adsp_cavs series, their flash script and log capturing script might be put into
+            # one script. According to twister's working flow, the script is better to be put in
+            # --device-serial-pty script in these cases. So like what we pass to --west-flash script,
+            # we also pass also build directory to pty script, in order to get more flexibility.
+            serial_cmd.append(self.build_dir)
```

Comment:
do we have document for this --device-serial-pty option? currently there are only one simple help in twister 
```
twister --device-testing
                        --device-serial-pty <script>
```
with this change the script shall take the build_dir as its first parameter, better to update this helper

Reply:
ok, @hakehuang, good suggestion. I could see where is the good place to add this.